sound/soc/sof/ipc4-topology.c

```diff
@@ -2147,8 +2151,14 @@ static int sof_ipc4_dai_config(struct snd_sof_dev *sdev, struct snd_sof_widget *
 		pipeline->skip_during_fe_trigger = true;
 		fallthrough;
 	case SOF_DAI_INTEL_ALH:
-		copier_data->gtw_cfg.node_id &= ~SOF_IPC4_NODE_INDEX_MASK;
-		copier_data->gtw_cfg.node_id |= SOF_IPC4_NODE_INDEX(data->dai_data);
+		/*
+		 * Do not clear the node ID during hw_free. It is needed to free the group_ida
```

Comment:
is this really during hw_free? Or did you mean widget_free? I don't see how hw_free and widget_unprepare are related?

Reply:
Widget free happens during hw_free or during suspend. Unprepare also happens during hw_free but it's just the for the AlH dais we free the group_ida is freed during unprepare and we should not clear the node ID until then

Reply:
ive changed the comment now.